File: JavaScript/06.js
/*
 * Crea un programa que se encargue de calcular el aspect ratio de una
 * imagen a partir de una url.
 * - Url de ejemplo:
 *   https://raw.githubusercontent.com/mouredevmouredev/master/mouredev_github_profile.png
 * - Por ratio hacemos referencia por ejemplo a los "16:9" de una
 *   imagen de 1920*1080px.
 */

import { Image } from "canvas";

function maximoComunDivisor(a, b) {
    return b === 0 ? a : maximoComunDivisor(b, a % b);
}

function aspectRatio(url){
    const img = new Image();
    img.src = url;
    img.onload = function() {
        const divisor = maximoComunDivisor(Math.round(img.width), Math.round(img.height))
        const ratioW = img.width / divisor;
        const ratioH = img.height / divisor;
        console.log(`Aspect ratio simplificado: ${ratioW}:${ratioH}`);
    }
}

aspectRatio("https://wallpapers.com/images/hd/1920x1080-aesthetic-glrfk0ntspz3tvxg.jpg");

File: Python/06.py
 #
 # Crea un programa que se encargue de calcular el aspect ratio de una
 # imagen a partir de una url.
 # - Url de ejemplo:
 #   https://raw.githubusercontent.com/mouredevmouredev/master/mouredev_github_profile.png
 # - Por ratio hacemos referencia por ejemplo a los "16:9" de una
 #   imagen de 1920*1080px.
 #

import requests
from PIL import Image
import io
import math

def aspectRatio(url):
    response = requests.get(url, stream=True)
    image_data = io.BytesIO(response.content)
    img = Image.open(image_data)
    width, height = img.size
    mcd = math.gcd(width, height)
    aspect_ratio = f"{width//mcd}:{round(height//mcd)}"
    return aspect_ratio

print(aspectRatio("https://pillow.readthedocs.io/en/stable/_images/masked_hopper.webp"))
print(aspectRatio("https://preview.redd.it/vo9vm1fcqrp71.jpg?auto=webp&s=cb4016edf50a37cf06dbe9e975ed9410b253bff0"))